Organizations

IJ-Lopez opened the pull request:
Falta la validación de administrador en el GET

IJ-Lopez committed:
Finished Organization Get Controller Method

Cree un DTO para enviar a través del método. Además, implementé "serializable" en la entidad de organización.

IJ-Lopez committed:
Created RequestOrganizationDTO

Cree un dto para recibir por parámetro en los request del controlador de organizaciones

IJ-Lopez committed:
Update OrganizationEntity.java

Modifique la clausula "where" para corregir el error en la sintaxis

IJ-Lopez committed:
Users Controller - Update + Delete (#16)

* Finished User Delete Controller Method

Tambien cree un DTO para enviar en la petición GET y cree un par de exception handlers para los NullPointerException (tirados por Lombok) y los UserServiceException (Que detectan cuando no se encontro el usuario deseado)

* Finished Update User Controller Method

Tambien modifique la entidad usuario para que el json ignore ciertos atributos, y modifique un método del servicio (put) que tenia un error en el manejo de la ID.

* User & Rol fix

Habia un problema en dichas entidades en las anotaciones para las queries SQL, ya fue corregido

* User Fix

Agregué un DTO para hacer el input de usuario (previamente habia modificado con @JsonIgnore los atributos que no queria recibir, ya revertí dichos cambios)

Pabpol committed:
Swagger configuration (#17)

* Swagger configuration

* pom.xml -> Spring Security dependencies taken out

IJ-Lopez committed:
Update Organization Controller Method created

Le falta la validación de administrador.
Cree un DTO para la petición.
También modifique el método update de la clase OrganizationService puesto que habia un error con el manejo de datos.

IJ-Lopez committed:
Merge branch 'develop' into organizations

nicotruk commented:
- El ticket del GET, puede no estar del todo claro, pero refiere a **una** Organización. Asique agregale que venga el id (como bien hiciste en el PATCH).
- Te pido si haces el POST de creacion que estuve revisando el backlog y no existe dicho ticket
- Vi la validacion del campo imagen. Esta bien pero fijate que cuando te manden algo que no cumpla, que el mensaje de la respuesta sea claro.

IJ-Lopez committed:
Finished Organization Controller

Cree los respectivos DTOs y reemplace el servicio por una interfaz

nicotruk commented:
- Email con formato invalido -> que el mensaje en response sea mas claro
- POST de creacion -> que no setee el updatedAt. Solamente createdAt.
- GET de id inexistente, devuelve 400 -> deberia ser 404. Y corregir mensaje
![Screenshot from 2021-12-03 10-12-20](https://user-images.githubusercontent.com/17879377/144608264-5eb64f0a-cd8a-48bc-8d9f-b43118e28532.png)

Yane05 committed:
Category endpoints (#18)

* Tickets 40 y 42

* Tickets 40 y 42

* application.properties -> No credentials

* ticket 41 y arreglo del 42

* Ticket 44 y correccion de 42

* Ticket 43

* application.properties -> No credentials

* arreglo de excepcion

* arreglo de excepcion

dellmdq committed:
Amazon S3 Bucket for upload and download images. (#21)

* Saving just in case

* FILE_NAME npe

* FILE_NAME npe

* S3 Post and Get Request Working

* probando git

* checking github config

* Git OK

* Just saving

* Save Fix. Now returns file path as a String

* Security Fix

* application.properties recovered

* pom.xml recovered

* S3Controller Swagger fix

* Deleted Comments

IJ-Lopez committed:
Exception handling fix

Arregle el manejo de excepciones. Modifiqué la clase servicio para que el `findById` devuelva `ParamNotFound`. También modifique de la clase "OrganizationEntity", el atributo `updated_at` para que no se persista en la creación, pero Spring lo devuelve de igual forma. Estoy trabajando en ello

IJ-Lopez committed:
Creation Post Fix

Arreglé el problema del updated_at

IJ-Lopez committed:
Merge branch 'develop' into organizations
